src/main/java/com/blackducksoftware/integration/hub/global/HubServerConfig.java

```diff
@@ -65,9 +75,12 @@ public void print(final IntLogger logger) {
         if (getHubUrl() != null) {
             logger.alwaysLog("--> Hub Server Url : " + getHubUrl());
         }
-        if (StringUtils.isNotBlank(getGlobalCredentials().getUsername())) {
+        if (getGlobalCredentials() != null && StringUtils.isNotBlank(getGlobalCredentials().getUsername())) {
             logger.alwaysLog("--> Hub User : " + getGlobalCredentials().getUsername());
         }
+        if (StringUtils.isNotBlank(apiKey)) {
+            logger.alwaysLog("--> Hub API Key : " + getApiKey());
```

Comment:
Is it OK (secure) to log the api key?

Reply:
We definitely shouldn't. Ryan left the following comment in hub-common-rest: 
"We don't log the headers here, as they contain a secret (the API key)" (in com.blackducksoftware.integration.hub.rest.ApiKeyRestConnection.clientAuthenticate()).

Reply:
I wouldn't log the API key being used. If we want to log maybe just log that an API key is being used. If we see Hub User we know we are dealing with a credentials rest connection.  If we see "Hub API Key Used" then we know we are using the APIKey rest connection.